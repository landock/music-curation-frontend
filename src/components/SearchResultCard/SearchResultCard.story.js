import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { linkTo } from '@storybook/addon-links';
import SearchResultCard from '.';

const storyProps = {
  trackName: 'whatever',
  artistName: 'Can',
  durationInSeconds: 3600,
  imageUrl: 'https://picsum.photos/150',
  streamUrl: '',
  recordLabels: ['Warner Music Group'],
};
storiesOf('SearchResultCard', module).add('default', () => (
  <SearchResultCard {...storyProps} />
));
