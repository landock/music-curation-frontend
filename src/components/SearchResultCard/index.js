import React from 'react';
import PropTypes from 'prop-types';

function SearchResultCard({
  trackName,
  artistName,
  durationInSeconds,
  imageUrl,
  streamUrl,
  recordLabels,
}) {
  return (
    <div style={{ display: 'flex' }}>
      <img style={{ height: '150px' }} src={imageUrl} />
      <div>
        <h4>
          {trackName}
          <button>add to curation</button>
        </h4>
        <p>{artistName}</p>
        <p>
          {recordLabels ? (
            recordLabels.map((label, index) => <span key={index}>{label}</span>)
          ) : (
            <span> no labels </span>
          )}
        </p>
        <p>{durationInSeconds}</p>
        <div>
          <button>tags</button>
          <button>Add to</button>
        </div>
      </div>
    </div>
  );
}

SearchResultCard.propTypes = {
  trackName: PropTypes.string,
  artistName: PropTypes.string,
  durationInSeconds: PropTypes.number,
  imageUrl: PropTypes.string,
  streamUrl: PropTypes.string,
  recordLabels: PropTypes.arrayOf(PropTypes.string),
};

export default SearchResultCard;
