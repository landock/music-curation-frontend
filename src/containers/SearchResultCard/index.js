import React from 'react';
import recycle from 'recycle';
import PropTypes from 'prop-types';
import { map, withLatestFrom } from 'rxjs/operators';

import { Item, Icon, Popup } from 'semantic-ui-react';

import AddTags from '../../components/AddTags';
import AddToCuration from '../../components/AddToCuration';
import {
  types as curationsTypes,
  actions as curationsActions,
} from '../../modules/CurrentCuration';
import { SortableElement, SortableHandle } from 'react-sortable-hoc';
import getSelectorStream from '../../getSelectorStream';

const SearchResultCard = recycle({
  displayName: 'SearchResultCard',
  dispatch(sources) {
    const buttonEventStream = getSelectorStream(
      sources.select,
      Icon,
      'onClick'
    );
    return [
      mapClickEventToActionWithPropsAsPayload(buttonEventStream, sources.props),
    ];
  },
  view(props, state) {
    return new SearchResultCardView(props);
  },
});

function mapClickEventToActionWithPropsAsPayload(eventStream, propsStream) {
  return eventStream.pipe(
    withLatestFrom(propsStream),
    map(([e, props]) => {
      return curationsActions.addCurationTrack({ ...props });
    })
  );
}

function SearchResultCardView({
  id,
  trackName,
  artistName,
  iconName,
  durationInSeconds,
  imageUrl,
  streamUrl,
  tags,
  recordLabels,
  curations,
}) {
  const DragHandleName = SortableHandle(({ trackName }) => {
    return <span>{trackName}</span>;
  });
  return (
    <Item>
      <Item.Image size="tiny" src={imageUrl} />
      <Item.Content>
        <Item.Header as="a">
          <DragHandleName trackName={trackName} />
          {': ' + id}
          <Icon name={iconName || 'plus'} link />
        </Item.Header>
        <Item.Description>{artistName}</Item.Description>
        <Item.Extra>
          {recordLabels ? (
            <span>{recordLabels.join(', ')}</span>
          ) : (
            <span> no labels </span>
          )}
          <p>{durationInSeconds}</p>
        </Item.Extra>
        <Popup
          trigger={
            <Icon style={{ cursor: 'pointer' }} size="small" name="tags" />
          }
          content={<AddTags tags={tags} />}
          on="click"
          position="top right"
          verticalOffset={-100}
          horizontalOffset={90}
          basic
        />
        <Popup
          trigger={<Icon name="share" style={{ cursor: 'pointer' }} />}
          content={<AddToCuration curations={curations} />}
          on="click"
          position="bottom right"
          verticalOffset={-133}
          horizontalOffset={-20}
          basic
        />
      </Item.Content>
    </Item>
  );
}

SearchResultCard.propTypes = {
  id: PropTypes.number,
  trackName: PropTypes.string,
  artistName: PropTypes.string,
  durationInSeconds: PropTypes.number,
  imageUrl: PropTypes.string,
  streamUrl: PropTypes.string,
  recordLabels: PropTypes.arrayOf(PropTypes.string),
};

const SortableSearchResultCard = SortableElement(props => (
  <SearchResultCard {...props} />
));
export {
  SortableSearchResultCard as default,
  mapClickEventToActionWithPropsAsPayload,
  SearchResultCardView,
};
