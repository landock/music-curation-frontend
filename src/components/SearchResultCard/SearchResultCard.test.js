
import React from 'react'
import { shallow, mount, render } from 'enzyme';
import renderer from 'react-test-renderer'
import SearchResultCard from '.'

it('SearchResultCard: default', () => {
  // const component = renderer.create(<SearchResultCard />)
  // const tree = component.toJSON()
  // expect(tree).toMatchSnapshot()
})
