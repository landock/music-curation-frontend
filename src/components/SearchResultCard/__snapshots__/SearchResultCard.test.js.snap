// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultCard: default 1`] = `
<div
  style={
    Object {
      "display": "flex",
    }
  }
>
  <img
    src={undefined}
    style={
      Object {
        "height": "150px",
      }
    }
  />
  <div>
    <h4>
      <button>
        add to curation
      </button>
    </h4>
    <p />
    <p>
      <span>
         no labels 
      </span>
    </p>
    <p />
    <div>
      <button>
        tags
      </button>
      <button>
        Add to
      </button>
    </div>
  </div>
</div>
`;
